### DISCUSSION TITLE
Is there any docs / tutorials on how to hook into electron for custom code?

### DISCUSSION THREAD

**Discussion by wardy484** [NativePHP/laravel#380] [Q&A] [answered]
I'm trying to launch applications on the host machine with custom arguments, for example on windows I might want to launch `cmd.exe /C ssh forge@some-ip`, this seems achievable from the electron node process but I'm unsure it's possible to hook into it without forking the whole electron plugin?

**Comment by simonhamp** [comment]
@wardy484 can you add a link to the specific part of the Electron docs you're referring to?

**Reply by wardy484** [reply]
More a stack overflow example than an electron thing. If wanted to have a play and test out the example where they used child_process instead of open external. So I was more curious if there was a method for me to hook into the JS process and run custom code. 

https://stackoverflow.com/questions/74814215/open-external-application-with-electron

**Reply by simonhamp** [reply]
Can't you do this directly from PHP using something like [`shell_exec`](https://www.php.net/manual/en/function.shell-exec.php)?

**Reply by wardy484** [reply]
I don't _think_ that will achieve the effect I'm after. So my intention is to be able to launch various applications, like a terminal window with an ssh session into a forge server, or a database connection in tables plus. I bit like how Laravel Herd does.

**Reply by simonhamp** [reply, marked as answer]
It should work...
![shell_exec](https://github.com/user-attachments/assets/9afd9b48-62a3-415c-ac89-30730b60fadf)

Anything you can run from the terminal can be run via `shell_exec`. There's nothing special about NativePHP that would prevent this

**Reply by wardy484** [reply]
Ah yep you're right. Turns out I'm just terrible at windows shell commands 😅  my session seemed to hang when I posted my previous comment but that's just because I was launching the command prompt within the php session instead of starting a new window! Thanks for this, I've learnt something new today!

For some reason in my head the php instance running inside php native was some sort of containerised instance 🤷‍♂️ no idea why.
